homework/index.js

```diff
@@ -1,47 +1,238 @@
       }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+    });
   }
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
+  function fillRepositoryDetailsTable(selectedRepository) {
+    const tableBody = document.querySelector('#table-body');
+
+    const repository = {
+      repository: selectedRepository.name,
+      description: selectedRepository.description,
+      forks: selectedRepository.forks,
+      updated: new Date(selectedRepository.updated_at).toLocaleString(),
+    };
+
+    Object.keys(repository).forEach((key, index) => {
+      const tr = Helper.createAndAppend('tr', tableBody);
+
+      if (index === 0) {
+        Helper.createAndAppend('td', tr, { class: 'label', text: key });
+        const td = Helper.createAndAppend('td', tr);
+        Helper.createAndAppend('a', td, {
+          href: selectedRepository.html_url,
+          target: '_blank',
+          text: repository[key],
+        });
       } else {
-        elem.setAttribute(key, value);
+        Helper.createAndAppend('td', tr, { class: 'label', text: key });
+        Helper.createAndAppend('td', tr, { text: repository[key] });
       }
     });
-    return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
+  async function fillContributorsList(selectedRepository) {
+    const list = document.querySelector('.contributors__list');
+    try {
+      const contributors = await Helper.fetchJSON(selectedRepository.contributors_url);
+      if (!contributors.length) {
+        Helper.createAndAppend('li', list, {
+          class: 'alert-warning',
+          text: 'There is no contributor for this repository!',
+        });
+        return;
+      }
+
+      contributors.forEach(contributor => {
+        const listItem = Helper.createAndAppend('li', list, {
+          class: 'contributors__item',
+        });
+        const listLink = Helper.createAndAppend('a', listItem, {
+          href: contributor.html_url,
+          target: '_blank',
+          class: 'contributors__link',
+        });
+        Helper.createAndAppend('img', listLink, {
+          class: 'contributors__avatar',
+          src: contributor.avatar_url,
+          alt: contributor.login,
+        });
+        Helper.createAndAppend('span', listLink, {
+          class: 'contributors__name',
+          text: contributor.login,
+        });
+        Helper.createAndAppend('span', listLink, {
+          class: 'contributors__badge',
+          text: contributor.contributions,
+        });
+      });
+    } catch (error) {
+      Helper.createAndAppend('li', list, { text: error.message, class: 'alert-error' });
+    }
+  }
+
+  class App {
+    constructor(select) {
```

Comment:
This constructor is never executed because you never create an instance of App (with `new App(select)`). Instead, you use the  `App` class as a `static` class.